velox/vector/DecodedVector.cpp

```diff
@@ -442,8 +442,9 @@ void DecodedVector::setBaseDataForBias(
 
 namespace {
 bool isWrapper(VectorEncoding::Simple encoding) {
-  return encoding == VectorEncoding::Simple::SEQUENCE ||
-      encoding == VectorEncoding::Simple::DICTIONARY;
+  return encoding == VectorEncoding::Simple::CONSTANT ||
```

Comment:
Can we improve `DecodedVector::wrap` here, such that if the encoding is `DICTIONARY(CONSTANT)` (or `DICTIONARY(DICTIONARY(CONSTANT))` etc.), we wrap the result with `CONSTANT` only?

Reply:
Yeah, that's doable, though we need to ensure the dictionary layer(s) doesn't introduce extra nulls. I'm curious how you see this may benefit the performance (or other aspects)?

Reply:
This will reduce the memory usage of downstream batches significantly